subprojects/snapshots/src/main/java/org/gradle/internal/vfs/WatchingVirtualFileSystem.java

```diff
@@ -28,21 +25,20 @@
 public interface WatchingVirtualFileSystem extends VirtualFileSystem {
 
     /**
-     * Start watching the known areas of the file system for changes.
+     * Called when the build is started and watching is disabled for the current build.
      *
-     * @param mustWatchDirectories directories that always should be watched even when not part of the VFS.
+     * This means that watchers should be teared down and no
+     * VFS state should be retained.
      */
-    void startWatching(Collection<File> mustWatchDirectories);
+    void watchingDisabledForCurrentBuild();
 
     /**
-     * Stop watching the known areas of the file system, and invalidate
-     * the parts that have been changed since calling {@link #startWatching(Collection)} ()}.
+     * Called when the build is started and watching is enabled for the current build.
      */
-    void stopWatching();
-
-    VirtualFileSystemStatistics getStatistics();
+    void afterStartingBuildWithWatchingEnabled();
 
-    interface VirtualFileSystemStatistics {
-        int getRetained(FileType fileType);
-    }
+    /**
+     * Called when the build is completed and watching is enabled for the current build.
+     */
+    void beforeCompletingBuildWithWatchingEnabled(File rootProjectDir);
```

Comment:
This sounds something like `updateWatchesAtEndOfBuild()`.

Reply:
I changed the names to make things clearer.